src/java/org/apache/cassandra/gms/Gossiper.java

```diff
@@ -2027,6 +2022,10 @@ public List<String> reloadSeeds()
         seeds.addAll(tmp);
         // Remove the old entries
         seeds.retainAll(tmp);
+
+        if (tmp.contains(getBroadcastAddressAndPort()))
+            seeds.remove(getBroadcastAddressAndPort());
```

Comment:
We can not remove that entry from tmp, because that would render tmp empty and we ask if tmp.isEmpty(), in that case we just return. But we do not want to return. Do you see? Because that would return prematurely and we would not change `seeds`.